app/assets/javascripts/index_map.js

```diff
@@ -45,15 +44,29 @@ function initMap() {
 
   var mc = new MarkerClusterer(map, unit_markers);
 
-  initHeatMap(map, heatMapData);
-
+  initHeatMap(map);
 }
 
-function initHeatMap(map, heatMapData) {
+function initHeatMap(map) {
+  var heatMapPoint;
+  var heatMapData = [];
+  var unit = [];
+  var monthly_rent;
+
+  for (var i in gon.rental_units) {
```

Comment:
Identifier 'rental_units' is not in camel case.<br>The body of a for in should be wrapped in an if statement to filter unwanted properties from the prototype.<br>'gon' is not defined.